### PR TITLE
Telemetry: Emit session_start during startup activities

### PR DESCRIPTION
**Pull request by samgst-amazon**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

"session_start: this metric is emitted upon starting of the toolkit. This could be because of update, reloads or installation."

Added a new routine to post-startup activities to emit session start before other tasks run.

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/AwsToolkitStartupMetrics.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.ProjectActivity
+import software.aws.toolkits.telemetry.SessionTelemetry
+import java.util.concurrent.atomic.AtomicBoolean
+
+class AwsToolkitStartupMetrics : ProjectActivity {
+    companion object {
+        private var runOnce = AtomicBoolean(false)
+    }
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/AwsToolkitStartupMetrics.kt:14]: take a look at `RunOnceUtil`

Comment (samgst-amazon) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/AwsToolkitStartupMetrics.kt]: didnt know that existed thanks

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/AwsToolkitStartupMetrics.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+package software.aws.toolkits.jetbrains.services.telemetry
+
+import com.intellij.ide.util.RunOnceUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.ProjectActivity
+import software.aws.toolkits.telemetry.SessionTelemetry
+
+class AwsToolkitStartupMetrics : ProjectActivity {
+    override suspend fun execute(project: Project) {
+        RunOnceUtil.runOnceForApp(taskId) {
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/AwsToolkitStartupMetrics.kt:13]: consider:
```suggestion
        RunOnceUtil.runOnceForApp(this::class.qualifiedName) {
```

**Review by rli — APPROVED** [review]

File: plugins/toolkit/jetbrains-core/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -195,6 +195,7 @@
         <statusBarWidgetFactory id="AwsSettingsPanel" implementation="software.aws.toolkits.jetbrains.core.credentials.AwsSettingsPanelInstaller"/>
         <statusBarWidgetFactory id="CawsSpaceProjectInfo" implementation="software.aws.toolkits.jetbrains.services.caws.CawsStatusBarInstaller"/>
 
+        <postStartupActivity implementation="software.aws.toolkits.jetbrains.services.telemetry.AwsToolkitStartupMetrics" order="first"/>
```
Comment (manodnyab) [inline plugins/toolkit/jetbrains-core/resources/META-INF/plugin.xml:198]: I wonder if we should skip the order here? Would it interfere with any other JB specific activities?

**Pull request merged**